core/src/env_platform/db_handler.bal

```diff
@@ -112,50 +112,50 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
     }
 }
 
-# The `updateApplicationDraft` function will alter the exsisting application draft with the incoming form details.
+# The `updateApplication` function will either alter the exsisting application draft with the incoming form details or 
+# add a new version of an application with the incoming form details.
 # 
 # + form - Form containing the tree removal data.
+# + applicationId - The Id of the application which should be altered.
 # + return - This function will return true if draft is updated in the database, false if not or else it returns a mongodb:Database error.
-function updateApplicationDraft(TreeRemovalForm form, string applicationId) returns boolean|error {
+function updateApplication(TreeRemovalForm form, string applicationId) returns boolean|error {
 
-    map<json> application = {
-        "title": form.title,
-        "applicationCreatedDate": {
-            "year": form.applicationCreatedDate.year,
-            "month": form.applicationCreatedDate.month,
-            "day": form.applicationCreatedDate.day,
-            "hour": form.applicationCreatedDate.hour,
-            "minute": form.applicationCreatedDate.minute
-        },
-        "removalDate": {
-            "year": form.removalDate.year,
-            "month": form.removalDate.month,
-            "day": form.removalDate.day,
-            "hour": form.removalDate.hour,
-            "minute": form.removalDate.minute
-        },
-        "reason": form.reason,
-        "applicationType": form.applicationType,
-        "requestedBy": form.requestedBy,
-        "permitRequired": form.permitRequired,
-        "landOwner": form.landOwner,
-        "treeRemovalAuthority": form.treeRemovalAuthority,
-        "city": form.city,
-        "district": form.district,
-        "nameOfTheLand": form.nameOfTheLand,
-        "planNumber": form.planNumber,
-        "area": extractAreaAsJSONArray(form.area),
-        "treeInformation": extractTreeInformationAsJSONArray(form.treeInformation)
-    };
+    map<json> application = constructApplicationVersion(form);
     log:printDebug("Constructed application: " + application.toString());
 
-    int|mongodb:DatabaseError updated = applicationCollection->update({"versions.0": application}, {"applicationId": applicationId});
+    int|mongodb:DatabaseError updated;
+    string applicationStatus = check getApplicationStatusByApplicationId(applicationId);
+
+    // If the exsiting application is a draft.
+    if (applicationStatus == "draft") {
+        if (form.applicationType == "draft") {
+            updated = applicationCollection->update({"versions.0": application}, {"applicationId": applicationId});
```

Comment:
since this updates the 0th index in versions array, does it mean that we are overriding whatever application is already in that index?

Reply:
Yes. We don't keep draft application versions. Then it is good to overriding the existing one. But when it comes to the submitted ones we have to make versions.

Reply:
That part is yet to be done right?

Reply:
It did't get it. The overall logic is 
* If the application with the given application ID is draft and incoming form is also a draft then, it will alter and add to the versions[0]. 
* If the application with the given application ID is draft and incoming form is submit then, it will alter and add to the versions[0] and change the status to draft.
* If the application with the giver application ID is submitted and the incoming form is submitted then, it will add the application to the versions array.
* else give appropriate errors.

Reply:
Right okay. My concern was whether we need to manage versions for applicatonStatus = draft and applicationType = submit. I guess that's not needed then.

Reply:
No it is not needed.